packages/legend-query-builder/src/stores/filter/QueryBuilderFilterState.ts

```diff
@@ -965,48 +1137,86 @@ export class QueryBuilderFilterState
     return Boolean(
       Array.from(this.nodes.values())
         .filter(filterByType(QueryBuilderFilterTreeConditionNodeData))
-        .map((node) => node.condition.value)
+        .map((node) =>
+          node.condition.rightConditionValue instanceof
+          FilterValueSpecConditionValueState
+            ? node.condition.rightConditionValue.value
+            : undefined,
+        )
         .filter(isNonNullable)
         .find((value) => isValueExpressionReferencedInValue(variable, value)),
     );
   }
 
+  isInvalidFilterPropertyExpressionState(
+    node: QueryBuilderFilterTreeNodeData,
+  ): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      !node.condition.propertyExpressionState.isValid
+    );
+  }
+
+  isInvalidValueSpecFilterValue(node: QueryBuilderFilterTreeNodeData): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      node.condition.rightConditionValue instanceof
+        FilterValueSpecConditionValueState &&
+      node.condition.rightConditionValue.value instanceof InstanceValue &&
+      !isValidInstanceValue(node.condition.rightConditionValue.value)
+    );
+  }
+
+  isInvalidPropertyExpressionStateFilterValue(
+    node: QueryBuilderFilterTreeNodeData,
+  ): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      node.condition.rightConditionValue instanceof
+        FilterPropertyExpressionStateConditionValueState &&
+      !node.condition.rightConditionValue.propertyExpressionState.isValid
+    );
+  }
+
   get allValidationIssues(): string[] {
     const validationIssues: string[] = [];
     Array.from(this.nodes.values()).forEach((node) => {
       if (node instanceof QueryBuilderFilterTreeConditionNodeData) {
-        if (
-          node.condition.value instanceof InstanceValue &&
-          !isValidInstanceValue(node.condition.value)
-        ) {
+        if (this.isInvalidValueSpecFilterValue(node)) {
           validationIssues.push(
             `Filter value for ${node.condition.propertyExpressionState.title} is missing or invalid`,
           );
         }
-        if (!node.condition.propertyExpressionState.isValid) {
+        if (this.isInvalidFilterPropertyExpressionState(node)) {
           validationIssues.push(
             `Derived property parameter value for ${node.condition.propertyExpressionState.title} is missing or invalid`,
           );
         }
+        if (
+          node.condition.rightConditionValue instanceof
+            FilterPropertyExpressionStateConditionValueState &&
+          this.isInvalidPropertyExpressionStateFilterValue(node)
+        ) {
+          validationIssues.push(
+            `Derived property parameter value for ${node.condition.rightConditionValue.propertyExpressionState.title} is missing or invalid`,
```

Comment:
would you mind explaining again why if the right hand of filter node is a property expression, then it is a `Derived property parameter value` ?